### PR TITLE
akka-http: HttpMessageParser swallows IllegalHeaderException #20272

### PR DESCRIPTION
**Pull request by pjazdzewski1990**
Catch clause in HttpMessageParser.parseBytes will intercept IllegalHeaderException being thrown and translate them into proper errors.
Refs https://github.com/akka/akka/issues/20272

### CODE REVIEW

**Comment by 2beaucoup** [comment]
The `Strict-Transport-Security` changes look unrelated.

**Comment by pjazdzewski1990** [comment]
@ktoso can you start CI on this PR?

**Comment by ktoso** [comment]
OK TO TEST

File: akka-http-core/src/test/scala/akka/http/scaladsl/model/headers/HeaderSpec.scala [hunk 1]
```diff
@@ -68,6 +68,12 @@ class HeaderSpec extends FreeSpec with Matchers {
         headers.`Strict-Transport-Security`.parseFromValueString("max-age=30; includeSubDomains") shouldEqual Right(headers.`Strict-Transport-Security`(30, true))
         headers.`Strict-Transport-Security`.parseFromValueString("max-age=30; includeSubDomains; preload") shouldEqual Right(headers.`Strict-Transport-Security`(30, true))
       }
+      "successful parse run with additional values" in {
+        headers.`Strict-Transport-Security`.parseFromValueString("max-age=30; includeSubDomains; preload; dummy") shouldEqual
+          Right(headers.`Strict-Transport-Security`(30, true))
+        headers.`Strict-Transport-Security`.parseFromValueString("max-age=30; includeSubDomains; dummy; preload") shouldEqual
+          Right(headers.`Strict-Transport-Security`(30, true))
+      }
```
Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/scaladsl/model/headers/HeaderSpec.scala]: these look unrelated

**Comment by ktoso** [comment]
LGTM

File: akka-http-core/src/test/scala/akka/http/scaladsl/model/headers/HeaderSpec.scala (hunk 1 above)
Comment (pjazdzewski1990) [inline akka-http-core/src/test/scala/akka/http/scaladsl/model/headers/HeaderSpec.scala]: @ktoso @2beaucoup Should I remove them in a follow-up PR?

Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/scaladsl/model/headers/HeaderSpec.scala]: I guess it doesn't hurt, those are quick. Was a bit confusing to see them here in the PR though
